File: coresuite-dotnet-standard-cs/reportwriter/ContactListReport.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.LayoutEngine;

namespace coresuite_dotnet_standard_cs.reportwriter
{
    public class ContactListReport
    {
        public static void Run(string outputPdfPath)
        {
            // Create the document's layout from a DLEX template
            DocumentLayout documentLayout = new DocumentLayout(Util.GetResourcePath("DLEX/ContactList.dlex"));

            NameValueLayoutData layoutData = new NameValueLayoutData();
            layoutData.Add("ContactGroups", ContactListExampleData.ContactGroups);

            // Layout the document using the parameters and set it's properties
            Document document = documentLayout.Layout(layoutData);
            document.Author = "DynamicPDF ReportWriter";
            document.Title = "Contact List Example";

            // Outputs the document to a file
            document.Draw(outputPdfPath);
        }
    }
}

File: coresuite-dotnet-framework-console-vb/reportwriter/ContactListReport.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.LayoutEngine

Public Class ContactListReport

    Public Shared Sub Run(outputPdfPath As String)

        ' Create the document's layout from a DLEX template
        Dim documentLayout As DocumentLayout = New DocumentLayout(Util.GetResourcePath("DLEX/ContactList.dlex"))

        Dim layoutData As NameValueLayoutData = New NameValueLayoutData()
        layoutData.Add("ContactGroups", ContactListExampleData.ContactGroups)

        ' Layout the document imports the parameters And set it's properties
        Dim document As Document = documentLayout.Layout(layoutData)
        document.Author = "DynamicPDF ReportWriter"
        document.Title = "Contact List Example"

        ' Outputs the document to a file
        document.Draw(outputPdfPath)
    End Sub
End Class

